api/src/main/scala/dx/api/DxApi.scala

```diff
@@ -951,23 +951,35 @@ case class DxApi(version: String = "1.0.0", dxEnv: DXEnvironment = DxApi.default
       }
 
     val allResults = workspaceResults ++ remaining.groupBy(_.project).flatMap {
-      case (None, files) if !inExecutionEnvironment && currentProjectId.isDefined =>
+      case (None, files) if currentProjectId.isDefined =>
         submitRequest(files.map(_.id).toSet, currentProject)
+      case (None, files) =>
+        throw new Exception(
+            s"""
+               |Platform wide file search is not recommended. Select a project by calling `dx select project-xxx`
+               |in command line or provide fully-qualified names to ${files.map(_.id)}
+               |""".stripMargin
+        )
       case (proj, files) =>
         submitRequest(files.map(_.id).toSet, proj)
     }
 
     if (validate) {
       val allResultsById = allResults.groupBy(_.id)
-      val multiple = allResultsById.filter(_._2.size > 1)
-      if (multiple.nonEmpty) {
-        throw new Exception(
-            s"One or more file IDs did not have a project specified and returned multiple search results: ${multiple}"
-        )
-      }
+      /*
+       * Here was also a validation that there is only one file with a given ID exists in the region, i.e. the file was
+       * not cloned to other projects. In APPS-1270 we removed this validation and only allowed describing files from
+       * other projects iff a fully qualified name was provided.
+       * */
```

Comment:
Also see Line 939 and https://github.com/dnanexus/dxCompiler/blob/c7b3061e49f5e7f413e0481fd880a8393655a256/executorCommon/src/main/scala/dx/executor/JobMeta.scala#L441
what shall we expect if we are referring to both `proj-A:file-X` and `proj-B:file-X`? Or all files are already cloned to the current workspace and this case would never happen?

Reply:
if a user provides `proj-A:file-X` and `proj-B:file-X` (file ID is the same) then [this line](https://github.com/dnanexus/dxScala/blob/7ac0ca7222b89d040f8f67d792eedde79970fec8/api/src/main/scala/dx/api/DxApi.scala#L939) will eliminate duplicates given conditionals are satisfied.

Reply:
might be related to APPS-1325 but nice work already! :D